### PR TITLE
Upgrade Kubebuilder dep from 3.4.1 to the commit 800fdeec6e5c

### PR DESCRIPTION
**Pull request by camilamacedo86**
### Description
- For Golang/Ansible/Helm language-based operators (go/v3, ansible/v1 and helm/v1): applying restrictive SCC for all containers scaffolded by the tool (`k8s versions < 1.19 will no longer work with`). ([More info](https://github.com/kubernetes-sigs/kubebuilder/pull/2700))
- For Golang-based language (go/v2) fix the issue introduced by removing the GO111MODULE=on from Dockerfile. ([More info](https://github.com/kubernetes-sigs/kubebuilder/pull/2678))
- For Golang/Ansible/Helm language-based operators (go/v3, ansible/v1 and helm/v1 language (go/v3), add a new comment with the option `leaderElectionReleaseOnCancel` ((More info)[https://github.com/kubernetes-sigs/kubebuilder/pull/2596])

### Goal
Only to get the changes from what we currently use in SDK from kubebuilder.

### Motivation:
- Ensure that scaffolds begin to be performed with a restrive Security Context
- Be able to do specific follow up PR so we can discuss properly how we will consume the new additions on Kubebuilder
- Make eaiser the process to keep SDK updated, otherwise, it is hard to properly review if we wait to do a big PR with all.
- Ensure that SDK features can work with the latest changes performed in Kubebuilder

**Extra info**
You can check the latest changes on Kubebuilder by looking: https://github.com/kubernetes-sigs/kubebuilder/compare/v3.4.1...master

Note that SDK users Kubebuilder as a LIB. So that all that is currently important is updated by default when we bump. However, all that was added to Kubebuilder but still not imported/used in SDK will not be added here. Therefore we can do follow up PRs to get the additions. Why? we have significant changes and additions for example adding the support for the phase 2 plugin, but if we try to do it all in the same PR it will be very hard to get properly reviewed.

### CODE REVIEW

File: go.mod
Comment (camilamacedo86) [inline go.mod:18]: @varshaprasad96 @ryantking  
Do we need to update these plugins first now before updating SDK?

Comment (camilamacedo86) [reply go.mod:18]: See: https://github.com/operator-framework/helm-operator-plugins/pull/169

**Comment by jmrodri** [comment]
@camilamacedo86 not sure if you want to include java in this PR or not. But v0.5.0 was released: https://github.com/operator-framework/java-operator-plugins/releases/tag/v0.5.0

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:5]:
```suggestion
      For Golang/Ansible/Helm/HybridHelm language-based operators (go/v3, ansible/v1, helm/v1, hybrid.helm/v1-alpha): applying restrictive SCC for all containers scaffolded by the tool (will not work with k8s versions < 1.19).
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:7]:
```suggestion
      ([More info](https://github.com/kubernetes-sigs/kubebuilder/pull/2700)). Important: You might be affected while updating Kubernetes to 1.25, Hence,  ensure that the containers
      used in your project are restrictive or defined with the property labels `pod-security.kubernetes.io` on clusters ([More info](https://kubernetes.io/docs/concepts/security/pod-security-admission/#pod-security-admission-labels-for-namespaces)). Therefore, we suggest
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:8]:
```suggestion
      that you begin to configure the projects according to the best practices. ([More info](https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/))
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:56]:
```suggestion
      **IMPORTANT NOTE**: If you are using `go/v2` we highly recommend to upgrade your project to latest `go/v3` plugin (default operator-sdk scaffold). Please be aware that
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:59]:
```suggestion
      Therefore, we are unable to upgrade this plugin continuously and solve tech-debts as it is
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:60]:
```suggestion
      very likely that it would soon be deprecated by the Kubebuilder community. Note that projects built
```

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:61]:
```suggestion
      with this plugin do not produce CRDs for example using v1 APIs so that cannot work on
```

Comment (varshaprasad96) [reply changelog/fragments/kb-800fdeec6e5c.yaml:61]: Just clarifying, do we mean to say that go/v2 plugins do not scaffold v1 CRDs. They support only v1beta1 which is deprecated and will be removed soon. Or am I getting it wrong?

File: changelog/fragments/kb-800fdeec6e5c.yaml
Comment (varshaprasad96) [inline changelog/fragments/kb-800fdeec6e5c.yaml:13]: Since this is a user-facing change in scaffolding, don't we need migration guide?

Comment (camilamacedo86) [reply changelog/fragments/kb-800fdeec6e5c.yaml:13]: is it about the pod restrictive? if yes, we added to see the migration steps.

File: go.mod
Comment (camilamacedo86) [reply go.mod:18]: solved

**Review by everettraven — APPROVED** [review]
This looks good to me but I think it would be best to have another set of eyes and approval before merging.

**Pull request merged**
